src/devices/CharacterLcd/LcdInterface.I2c.cs

```diff
@@ -115,6 +117,31 @@ public override void SendData(ReadOnlySpan<byte> values)
                     _device.Write(buffer.Slice(0, currentValues.Length + 1));
                 }
             }
+
+            public override void SendData(ReadOnlySpan<char> values)
+            {
+                // There is a limit to how much data the controller can accept at once. Haven't found documentation
+                // for this yet, can probably iterate a bit more on this to find a true "max". 40 was too much.
+                const int MaxCopy = 20;
+                Span<byte> buffer = stackalloc byte[MaxCopy + 1];
+                buffer[0] = (byte)ControlByteFlags.RegisterSelect;
+                Span<byte> bufferData = buffer.Slice(1);
+
+                while (values.Length > 0)
+                {
+                    ReadOnlySpan<char> buff = values.Slice(0, values.Length > MaxCopy ? MaxCopy : values.Length);
+                    // As we are in a while loop, we can't use stackalloc
+                    Span<byte> currentValues = new byte[buff.Length];
+                    for (int i = 0; i < buff.Length; i++)
+                    {
+                        currentValues[i] = (byte)buff[i];
```

Comment:
you seem to be copying these without checking the range, you might get some characters randomly. Wouldn't it make more sense to ask people wanting to call this to use Encoding.ASCII.GetBytes or similar instead of this new method?

Reply:
The all purpose of LcdCharacterEncoding is to do this job. We add additional characters, not aligned with the traditional ASCII way. The class is a acutally doing this. It was already existing and working code, adjusted the definition only from byte to char to be compatible with devices having larger capacities than 255 characters

Reply:
should we remove byte overload in that case?

Reply:
The byte one is the one used directly in the small LCD, the byte ones avoid having to reallocate a new buffer and avoid some over complicated part on the interface. @pgrawehr your feedback here would help as we chatted about this over the weekend.

Reply:
I would keep it, as it is used if the LcdInterface is used directly or if the character translation is already done. The char overload is only to prevent another buffer copy in case we do a char->char mapping instead.

Reply:
but why can't char overload provide sane default implementation and be overriden only if it makes sense?